nodes/src/nodes/llm_base/resilience.py

```
@@ -0,0 +1,422 @@
+        """
+        if not self._breaker.allow_request():
+            raise CircuitBreakerOpenError(self.provider_name, self._breaker.recovery_remaining())
+
+        @retry_with_backoff(max_retries=self.max_retries, base_delay=self.base_delay, max_delay=self.max_delay, retryable_types=self.retryable_types)
+        def _inner() -> Any:
+            return fn(*args, **kwargs)
+
+        try:
+            result = _inner()
+            self._breaker.record_success()
+            return result
```

Ruff (0.15.7)

[warning] Consider moving this statement to an `else` block

(TRY300)